source/_posts/vue-design-patterns.md

```diff
@@ -0,0 +1,982 @@
+    
+    export interface VisualizationPreferences {
+      darkMode: boolean;
+      borders: "square" | "rounded";
+    }
+    
+    export const VISUALIZATION_PREFERENCES: InjectionKey<
+      VisualizationPreferences | any
+    > = Symbol("visualizationPreferences");
+```
+
+The next component implemented is `OptionList.vue`, which works as a Presentational Component (only renders every **OptionItem** component if it has the necessary data, it is only a dummy component).
+
+```ts
+    // File: OptionList.vue
+    
+    <template>
+      <div v-for="({ title, image }, index) in data" :key="index">
+        <OptionItem :title="title" :image="image" />
+      </div>
+    </template>
+    
+    <script lang="ts">
+    import { defineComponent, type PropType } from "vue";
+    import OptionItem, { type OptionProps } from "./OptionItem.vue";
+    
+    export default defineComponent({
+      components: { OptionItem },
+      props: {
+        data: {
+          type: Array as PropType<OptionProps[]>,
+          default: () => [],
+        },
+      },
+      setup() {
+        return {};
+      },
+    });
+    </script>
+```
+
+Now, the `OptionContainer.vue` is the Provider. It contains the data to pass to `OptionItem.vue` through the `provide()` method, as we can see below.
+
+```ts
+    // File: OptionContainer.vue
+    
+    <script lang="ts">
+    import { computed, defineComponent, provide, reactive } from "vue";
+    import type { OptionProps } from "./OptionItem.vue";
+    import OptionList from "./OptionList.vue";
+    import {
+      VISUALIZATION_PREFERENCES,
+      type VisualizationPreferences,
+    } from "./OptionSymbols";
+    
+    export default defineComponent({
+      components: { OptionList },
+      setup() {
+        const preferences = reactive<VisualizationPreferences>({
+          borders: "square",
+          darkMode: false,
+        });
+        const data = reactive<OptionProps[]>([
+          {
+            title: "Nunc massa ex, vulputate id tincidunt",
+            image: "https://random.imagecdn.app/500/200#1",
+          },
+          {
+            title: "Donec facilisis, mauris a vulputate",
+            image: "https://random.imagecdn.app/500/200#12",
+          },
+          {
+            title: " Curabitur luctus mollis aliquam",
+            image: "https://random.imagecdn.app/500/200#33",
+          },
+        ]);
+        // The important part 👇
+        provide(VISUALIZATION_PREFERENCES, {
+          darkMode: computed(() => preferences.darkMode),
+          borders: computed(() => preferences.borders),
+        });
+    
+        return { data, preferences };
+      },
+    });
+    </script>
+```
+
+The code is large (this is the script part only). The `provide()` method receives the same `VISUALIZATION\_PREFERENCES` seen previously as the first parameter, and the data to be injected as a second parameter (we are using `computed()` to turn reactive that data).
```

Comment:
Same here a slash \ in `VISUALIZATION\_PREFERENCES`

Reply:
modified and done, thanks